alts/build.gradle

```diff
@@ -13,35 +13,35 @@ sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 dependencies {
-    compile project(':grpc-auth'),
-            project(':grpc-core'),
+    api project(':grpc-core')
+    compileOnly libraries.javax_annotation
+    implementation project(':grpc-auth'),
             project(':grpc-grpclb'),
             project(':grpc-protobuf'),
             project(':grpc-stub'),
             libraries.lang,
-            libraries.protobuf,
             libraries.conscrypt
-    def nettyDependency = compile project(':grpc-netty')
-    compile (libraries.google_auth_oauth2_http) {
+    def nettyDependency = implementation project(':grpc-netty')
+    implementation (libraries.google_auth_oauth2_http) {
         // prefer our own versions instead of google-auth-oauth2-http's dependency
         exclude group: 'com.google.guava', module: 'guava'
         // we'll always be more up-to-date
         exclude group: 'io.grpc', module: 'grpc-context'
     }
-    compileOnly libraries.javax_annotation
-
-    shadow configurations.compile.getDependencies().minus(nettyDependency)
-    shadow project(path: ':grpc-netty-shaded', configuration: 'shadow')
-
-    testCompile project(':grpc-testing'),
+    guavaDependency 'implementation'
+    testImplementation project(':grpc-testing'),
             project(':grpc-testing-proto'),
             libraries.guava,
             libraries.guava_testlib,
             libraries.junit,
             libraries.mockito,
             libraries.truth
-    testRuntime libraries.netty_tcnative,
-            libraries.netty_epoll
+    testRuntimeOnly libraries.netty_epoll,
```

Comment:
Note: unnecessary changes like reordering deps and moving the type of deps (like `shadow` here) around made reviewing this PR significantly harder. Sorting dependencies is nice, but would have been much better as a separate PR.

Reply:
Sorry for the noise and extra time and effort spent for the review because of it. I'll send smaller PRs without unnecessary changes like reordering deps.